Allow HTTP POST calls for any showcase PATCH RPCs

I'm trying to add in the showcase compliance test suite for Java and I'm running into an issue with the PATCH calls. 
Error:
```
Caused by: com.google.api.client.http.HttpResponseException: 400 Bad Request
POST http://localhost:7469/v1beta1/repeat:bodypatch
{"error":{"code":400,"message":"unrecognized request: POST \"/v1beta1/repeat:bodypatch\"","details":null,"Body":"","Header":null,"Errors":null}}
	at com.google.api.client.http.HttpResponseException$Builder.build(HttpResponseException.java:293)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1118)
	at com.google.api.gax.httpjson.HttpRequestRunnable.run(HttpRequestRunnable.java:118)
	... 6 more
```

Java Gax-httpjson's module sends PATCH requests as a POST request with a `x-http-method-override: PATCH` header, which results in a 400 Bad Request. Looking at the historical reasons for this: 
https://github.com/googleapis/gapic-generator-java/blob/c23f981e2ac3c573bed51e725dc7061551179400/gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpRequestRunnable.java#L217-L226
```
    // A workaround to support PATCH request. This assumes support of "X-HTTP-Method-Override"
    // header on the server side, which GCP services usually do.
    //
    // Long story short, the problems is as follows: gax-httpjson depends on NetHttpTransport class
    // from google-http-client, which depends on JDK standard java.net.HttpUrlConnection, which does
    // not support PATCH http method.
    //
    // It is a won't fix for JDK8: https://bugs.openjdk.java.net/browse/JDK-8207840.
    // A corresponding google-http-client issue:
    // https://github.com/googleapis/google-http-java-client/issues/167
```

I believe the mux router setup for bodyPatch to expect a PATCH request https://github.com/googleapis/gapic-showcase/blob/b94ecfc51059b49770e5bdb6f0d7ea07903158e8/server/genrest/genrest.go#L43

Possible fix (Thanks to Noah for pointing me here):
- Adding a conditional for PATCH here: https://github.com/googleapis/gapic-showcase/blob/b94ecfc51059b49770e5bdb6f0d7ea07903158e8/util/genrest/goviewcreator.go#L292-L294
so it would output
`router.HandleFunc("/v1beta1/repeat:bodypatch", rest.HandleRepeatDataBodyPatch).Methods("PATCH", "POST")`

Alternatives:
- Does Go Mux support `x-http-method-override`?

--- a/util/genrest/goviewcreator.go
+++ b/util/genrest/goviewcreator.go
@@ -290,7 +290,12 @@ func NewView(model *gomodel.Model) (*goview.View, error) {
 	// add '\n' to the "ExtremeValues" ComplianceGroup in compliance_suite.json.
 
 	for _, handler := range registered {
-		file.P(`  router.HandleFunc(%q, rest.%s).Methods(%q)`, handler.pattern, handler.function, handler.verb)
+		// Java's PATCH requests are sent as POST requests with `x-http-methods-override` header
+		if handler.verb == "PATCH" {
+			file.P(`  router.HandleFunc(%q, rest.%s).Methods(%q, %q)`, handler.pattern, handler.function, handler.verb, "POST")
+		} else {
+			file.P(`  router.HandleFunc(%q, rest.%s).Methods(%q)`, handler.pattern, handler.function, handler.verb)
+		}
 	}
 	file.P(`  router.PathPrefix("/").HandlerFunc(rest.catchAllHandler)`)
 	file.P(`}`)

--- a/util/genrest/goviewcreator.go
+++ b/util/genrest/goviewcreator.go
@@ -290,11 +290,11 @@ func NewView(model *gomodel.Model) (*goview.View, error) {
 	// add '\n' to the "ExtremeValues" ComplianceGroup in compliance_suite.json.
 
 	for _, handler := range registered {
-		// Java's PATCH requests are sent as POST requests with `x-http-methods-override` header
+		file.P(`  router.HandleFunc(%q, rest.%s).Methods(%q)`, handler.pattern, handler.function, handler.verb)
+		// Java's PATCH requests are sent as POST requests with a `X-HTTP-Method-Override` header
 		if handler.verb == "PATCH" {
-			file.P(`  router.HandleFunc(%q, rest.%s).Methods(%q, %q)`, handler.pattern, handler.function, handler.verb, "POST")
-		} else {
-			file.P(`  router.HandleFunc(%q, rest.%s).Methods(%q)`, handler.pattern, handler.function, handler.verb)
+			// HeadersRegexp is used to match for both POST Http Verb and the Header value
+			file.P(`  router.HandleFunc(%q, rest.%s).HeadersRegexp("X-HTTP-Method-Override", "^PATCH$").Methods("POST")`, handler.pattern, handler.function)
 		}
 	}
 	file.P(`  router.PathPrefix("/").HandlerFunc(rest.catchAllHandler)`)

